sys-utils/setarch.c

```diff
@@ -293,6 +334,57 @@ static void verify_arch_domain(struct arch_domain *doms, struct arch_domain *tar
 	errx(EXIT_FAILURE, _("Kernel cannot set architecture to %s"), wanted);
 }
 
+static const struct { int value; const char * const name; } all_personalities[] = {
+#define X(opt) { .value = opt, .name = #opt },
+	ALL_PERSONALITIES
+#undef X
+};
+
+static const struct { int value; const char * const name; } all_options[] = {
+#define X(opt) { .value = opt, .name = #opt },
+	ALL_OPTIONS
+#undef X
+};
+
+static void show_current_personality(void)
+{
+	unsigned long options;
+	size_t i;
+	int pers = personality(0xffffffff);
+	if (pers == -1)
+		errx(EXIT_FAILURE, _("Can not get current kernel personality: %s"), strerror(errno));
```

Comment:
It seems better to use `err(EXIT_FAILURE, _("Can not get current kernel personality");` rathe rather than errx() + strerror().